libs/blocks/commerce/commerce.js

```diff
@@ -0,0 +1,139 @@
+  const { env, locale } = getConfig();
+  const {
+    literalScriptUrl,
+    scriptUrl,
+    country,
+    language,
+    tacocatEnv,
+  } = getTacocatEnv(env.name, locale);
+
+  loadScript(literalScriptUrl)
+    .catch(() => ({})) /* ignore if literals fail */
+    .then(() => loadScript(scriptUrl))
+    .then(() => {
+      runTacocat(tacocatEnv, country, language);
+      resolve();
+    });
+});
+
+export const formatOfferDetails = (o) => {
+  const formattedOffer = {};
+
+  function formatPrice(price, format) {
+    const currency = format.match(/'([^']+)'/);
+    return `${currency[1]}${price}`;
+  }
+
+  const {
+    offerType,
+    offerId,
+    productArrangementCode,
+    pricePoint,
+    customerSegment,
+    commitment,
+    term,
+    offerSelectorIds,
+    priceDetails,
+  } = o;
+
+  formattedOffer.offerType = offerType;
+  formattedOffer.offerId = offerId;
+  formattedOffer.productArrangementCode = productArrangementCode;
+  formattedOffer.pricePoint = pricePoint;
+  formattedOffer.customerSegment = customerSegment;
+  formattedOffer.commitment = commitment;
+  formattedOffer.term = term;
+  formattedOffer.offerSelectorIds = offerSelectorIds;
+  formattedOffer.price = formatPrice(priceDetails.price, priceDetails.formatString);
+  return formattedOffer;
+};
+
+export const decorateOfferDetails = async (el, of, searchParams) => {
+  function formatOfferDetailKeys(str) {
+    const details = str.split(/(?=[A-Z])/);
+    const allCapsDetail = details.map((detail) => detail.toUpperCase());
+    const result = allCapsDetail.join(' ');
+    return result;
+  }
+  const offerDetailsList = document.createElement('ul');
+  offerDetailsList.className = 'offer-details';
+  const offer = formatOfferDetails(of);
+  Object.entries(offer).forEach(([key, value]) => {
+    const offerData = document.createElement('li');
+    const offerKey = document.createElement('span');
+    const offerValue = document.createElement('span');
+    offerData.classList.add('offer-detail');
+    offerKey.classList.add('offer-key');
+    offerValue.classList.add('offer-value');
+    offerKey.textContent = `${formatOfferDetailKeys(key)}: `;
+    offerValue.textContent = value;
+    offerData.appendChild(offerKey);
+    offerData.appendChild(offerValue);
+    offerDetailsList.appendChild(offerData);
+  });
+  const checkoutLink = document.createElement('a');
+  checkoutLink.textContent = 'Checkout link';
+  const perpetual = searchParams.get('perp') === 'true' || undefined;
+  const promotionCode = searchParams.get('promo') || undefined;
+  const options = omitNullValues({
+    perpetual,
+    promotionCode,
+    wcsOsi: searchParams.get('osi'),
+    ...getCheckoutContext(searchParams, getConfig()),
+  });
+  const checkoutUrl = buildCheckoutButton(checkoutLink, options);
+  checkoutUrl.target = '_blank';
+  offerDetailsList.appendChild(checkoutUrl);
+  el.append(offerDetailsList);
+};
+
+export const handleSearch = async (e, el) => {
```

Comment:
nit: suggestion to rename `e` to `event`
`async (event, el) => {`

Reply:
Done ✅